### PR TITLE
DOC-1754: Advanced Typography plugin documentation

### PR DESCRIPTION
**Pull request by brian-forte**
Related Ticket: [DOC-1754](https://ephocks.atlassian.net/browse/DOC-1754)

Description of Changes:
* Document the Advanced Typography plugin

Pre-checks:
- [ ] Branch prefixed with `feature/6/` or `hotfix/6/`
- [ ] `modules/ROOT/nav.adoc` has been updated (if applicable)
- [ ] Files has been included where required (if applicable)
- [ ] Files removed have been deleted, not just excluded from the build (if applicable)
- [ ] (New product features only) Release Note added

Review:
- [ ] Documentation Team Lead has reviewed

### CODE REVIEW

File: modules/ROOT/pages/available-menu-items.adoc [hunk 1]
```diff
@@ -37,6 +37,11 @@ include::partial$misc/plugin-menu-item-id-boilerplate.adoc[]
 :plugincode: advtable
 include::partial$misc/plugin-menu-item-id-boilerplate.adoc[]
 
+:plugincategory: premium
+:pluginname: Advanced Typography
+:plugincode: typography
+include::partial$misc/plugin-menu-item-id-boilerplate.adoc[]
```
Comment (tiny-ben-tran) [inline modules/ROOT/pages/available-menu-items.adoc:43]:
```suggestion
:pluginpage: advanced-typography.adoc
include::partial$misc/plugin-menu-item-id-boilerplate.adoc[]
```

Either use the above suggestion or rename `advanced-typopgrahy.adoc` to `typopgrahy.adoc`.

File: modules/ROOT/pages/available-toolbar-buttons.adoc [hunk 2]
```diff
@@ -37,6 +37,11 @@ include::partial$misc/plugin-toolbar-button-id-boilerplate.adoc[]
 :plugincode: advtable
 include::partial$misc/plugin-toolbar-button-id-boilerplate.adoc[]
 
+:plugincategory: premium
+:pluginname: Advanced Typography
+:plugincode: typography
+include::partial$misc/plugin-toolbar-button-id-boilerplate.adoc[]
```
Comment (tiny-ben-tran) [inline modules/ROOT/pages/available-toolbar-buttons.adoc:43]:
```suggestion
:pluginpage: advanced-typography.adoc
include::partial$misc/plugin-toolbar-button-id-boilerplate.adoc[]
```
Same thing here

File: modules/ROOT/partials/commands/typography-cmds.adoc [hunk 3]
```diff
@@ -0,0 +1,11 @@
+[cols="1,3",options="header"]
+|===
+|Command |Description
+|mceTypography |[ TO BE ADDED ].
```
Comment (tiny-ben-tran) [inline modules/ROOT/partials/commands/typography-cmds.adoc:4]: @brian-forte, you or I can fill these `TO BE ADDED` later

File: modules/ROOT/pages/advanced-typography.adoc [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+  selector: 'textarea',  // change this value according to your HTML
+  plugins: '{plugincode}',
+  toolbar: '{plugincode}',
+  language: 'en_US' // This is required to configure the default language used by typography.
+});
+----
+
+// Delete everything not require/relevant from this point on.
+
+== Options
```
Comment (ShiridiGandham) [inline modules/ROOT/pages/advanced-typography.adoc:45]: @brian-forte Below options needs to be documented -

- typography_default_lang
- typography_langs

Test Fiddle used by QA team  for reference  - https://fiddle.tiny.cloud/uqiaab/58

**Review by TheSpyder — CHANGES_REQUESTED** [review]
This was created from release and then had staging merged into it, so it looks like there's a bunch of unnecessary stuff in the PR

**Comment by ephox-mogran** [comment]
Closed in favour of the rebased: https://github.com/tinymce/tinymce-docs/pull/2578

File: modules/ROOT/examples/live-demos/typography/index.js [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+  'en-GB': 'English (UK)',
+  'bg': 'Bulgarian',
+  'be': 'Belarusian',
+  'ca': 'Catalan',
+  'cs': 'Czech',
+  'ga': 'Irish',
+  'da': 'Danish',
+  'nl': 'Dutch',
+  'fi': 'Finnish',
+  'fr': 'French',
```
Comment (Skylite73) [inline modules/ROOT/examples/live-demos/typography/index.js:64]: Should we also temporarily remove French from here?

Comment (ephox-mogran) [reply modules/ROOT/examples/live-demos/typography/index.js:64]: This PR was closed in favour of https://github.com/tinymce/tinymce-docs/pull/2578, so you'll need to review that one instead.

File: modules/ROOT/pages/a11ychecker.adoc [hunk 6]
```diff
@@ -357,7 +357,7 @@ editor.plugins.a11ychecker.toggleaudit();
 
 Conducts an accessibility audit and reports the results without opening the dialog. The report produced is an array of objects, where each object represents an xref:a11ychecker.adoc#issue[issue].
 
-==== Example: using `+getReport()+`
+==== Example: Using `+getReport()+`
```
Comment (Skylite73) [inline modules/ROOT/pages/a11ychecker.adoc:360]: Bit strange to have in `typography` doc PR but OK.

File: modules/ROOT/partials/configuration/advanced-typography.adoc [hunk 7]
```diff
@@ -0,0 +1,205 @@
+[[content_langs]]
+== `+content_langs+`
```
Comment (Skylite73) [inline modules/ROOT/partials/configuration/advanced-typography.adoc:2]: You may be thinking of `typography_default_langs` which has this functionality. [`content_langs` already defined](https://github.com/tinymce/tinymce-docs/blob/86adeabb818e5270cc2ce19f5dfd31ff67ed30e8/modules/ROOT/partials/configuration/content_langs.adoc)

File: modules/ROOT/pages/advanced-typography.adoc [hunk 8]
```diff
@@ -0,0 +1,60 @@
+= Advanced Typography plugin
+:navtitle: Advanced Typography
+:description: Automatically or interactively replace typewriter-centric characters with characters preferred by typesetters.
+:description_short: Auomatically or interactively typeset {productname} documents.
```
Comment (Skylite73) [inline modules/ROOT/pages/advanced-typography.adoc:4]: No automatic behaviour by default. Integrators would have to implement this themselves if they want such a feature.

File: modules/ROOT/pages/advanced-typography.adoc [hunk 9]
```diff
@@ -0,0 +1,60 @@
...
+:plugincode: typography
+:plugincategory: premium
+
+include::partial$misc/admon-premium-plugin.adoc[]
+
+include::partial$misc/admon-requires-6.3v.adoc[]
+
+The {pluginname} plugin, depending on its setting, automatically or interactively replaces common typewriter-style characters with their professional typesetter’s equivalents.
+
+For example, if English-language typography rules are set, the {pluginname} plugin can automatically replace the double-hyphen (`--`) with an em-dash (`—`). And it can automatically replace the teardrop apostrophe (`'`) with opening and closing quote marks (`‘` and `’`).
```
Comment (Skylite73) [inline modules/ROOT/pages/advanced-typography.adoc:16]: Be careful when we say 'English-language'. The `&quot;` replacement mentioned here is for `en_UK` whereas `en_US`, the default option, replaces `&quot;` with `"`.

File: modules/ROOT/partials/toolbar-button-ids/typography-toolbar-buttons.adoc [hunk 10]
```diff
@@ -0,0 +1,5 @@
+[cols="1,3",options="header"]
+|===
+|Toolbar button identifier |Description
+|`+typopgrahy+` | [ TO BE ADDED ]
```
Comment (ShiridiGandham) [inline modules/ROOT/partials/toolbar-button-ids/typography-toolbar-buttons.adoc:4]: Typo - typopgrahy -> typography

**Pull request closed without merging**